spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageUtils.java

```diff
@@ -112,8 +112,8 @@ else if (MessageHeaders.ID.equals(messageAttribute.getKey())) {
 				messageHeaders.put(MessageHeaders.ID,
 						UUID.fromString(messageAttribute.getValue().getStringValue()));
 			}
-			else if (MessageAttributeDataTypes.STRING
-					.equals(messageAttribute.getValue().getDataType())) {
+			else if (messageAttribute.getValue().getDataType()
+					.startsWith(MessageAttributeDataTypes.STRING)) {
```

Comment:
let's change it to `MessageAttributeDataTypes.STRING.startsWith(messageAttribute.getValue().getDataType())`

Reply:
getDataType may return „String” or string with label which is: „String.<type extensions labels>„

MessageAttributeDataTypes.STRING is always „String” 

If we replace as You suggest we will get:
„String”.startsWith(„String.<Here goes the label>”) which would be rejected, but we want to support everything which starts from „String”

Reply:
it was my fault, please ignore these comments. You are right.